pyproject.toml

```diff
@@ -141,12 +141,12 @@ features = [
 ]
 
 [tool.black]
-target-version = ["py39"]
+target-version = ["py310"]
```

Comment:
Now that we dropped support for py39, we can use `X | None` rather than `typing.Optional[X]` since it is natively supported in the language starating with python 3.10. See the build failures.